src/charm.py

```diff
@@ -121,6 +124,34 @@ def _is_legacy_v1_relation(self, relation):
 
         return True
 
+    def _configure_ingress(self, event: HookEvent) -> None:
+        """Set up ingress if a relation is joined, config changed, or a new leader election.
+
+        Also since :class:`TraefikRouteRequirer` may not have been constructed with an existing
+        relation if a :class:`RelationJoinedEvent` comes through during the charm lifecycle, if we
+        get one here, we should recreate it, but OF will give us grief about "two objects claiming
+        to be ...", so manipulate its private `_relation` variable instead.
+
+        Args:
+            event: a :class:`HookEvent` to signal a change we may need to respond to.
+        """
+        if not self.unit.is_leader():
+            return
+
+        # If it's a RelationJoinedEvent, set it in the ingress object
+        if isinstance(event, RelationJoinedEvent):
+            self._ingress._relation = event.relation
```

Comment:
it's very weird that it wasn't there to begin with. Where have you being seeing this? 
If it's Harness, this is an artifact due to the charm not being reinitialized on each event and should be fixed in test code, not charm code.

Reply:
That is based on a similar issue in grafana: https://github.com/canonical/grafana-k8s-operator/blob/main/src/charm.py#L353

Reply:
Yup, seems like it was a testing artifact. Removed it now.